### PR TITLE
ssh-vault 1.0.1

### PR DESCRIPTION
**Pull request by nbari**
- [x ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/ssh-vault.rb [hunk 1]
```diff
@@ -18,18 +18,26 @@ class SshVault < Formula
...
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "openssl@3"
+  end
 
   def install
-    ldflags = "-s -w -X main.version=#{version}"
-    system "go", "build", *std_go_args(ldflags: ldflags), "cmd/ssh-vault/main.go"
+    system "cargo", "install", *std_cargo_args
```
Comment (chenrui333) [inline Formula/s/ssh-vault.rb:29]:
```suggestion
    # Ensure that the `openssl` crate picks up the intended library.
    ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
    ENV["OPENSSL_NO_VENDOR"] = "1"

    system "cargo", "install", *std_cargo_args
```

File: Formula/s/ssh-vault.rb
Comment (chenrui333) [inline Formula/s/ssh-vault.rb:1]:
```
  def check_binary_linkage(binary, library)
    binary.dynamically_linked_libraries.any? do |dll|
      next false unless dll.start_with?(HOMEBREW_PREFIX.to_s)

      File.realpath(dll) == File.realpath(library)
    end
  end

  test do
...

    [
      Formula["openssl@3"].opt_lib/shared_library("libssl"),
      Formula["openssl@3"].opt_lib/shared_library("libcrypto"),
    ].each do |library|
      assert check_binary_linkage(bin/"ssh-vault", library),
            "No linkage with #{library.basename}! Cargo is likely using a vendored version."
    end
  end
```

**Comment by chenrui333** [comment]
also you need squash the commtis into one as `ssh-vault 0.12.11`

File: Formula/s/ssh-vault.rb
Comment (chenrui333) [reply Formula/s/ssh-vault.rb:1]: note, this would be linux only

File: Formula/s/ssh-vault.rb [hunk 2]
```diff
@@ -18,18 +18,30 @@ class SshVault < Formula
...
-    fingerprint = output.split("\n").first.split(";").last
-    cmd = "#{bin}/ssh-vault -k https://ssh-keys.online/key/#{fingerprint} view"
-    output = pipe_output(cmd, output, 0)
+    output = pipe_output("#{bin}/ssh-vault create -u new", "hi")
+    data = output.match(/-{3,}\n(.*?)-{3,}\n(.*)/m)
+    extracted_content = data[1].strip if data
+    parts = extracted_content.split("|").map(&:strip)
+    vault = parts[0].match(/"([^"]*)"/)
+    vault = vault[1].strip if vault
+    cmd = parts[1]
```
Comment (chenrui333) [inline Formula/s/ssh-vault.rb:43]: this seems a bit complicated 😅

**Comment by nbari** [comment]
Hi @chenrui333 I bumped version to `1.0.1` but why does this PR keep building with `0.12.11`, closing and creating a new PR could fix this?

**Comment by chenrui333** [comment]
one sec

**Comment by chenrui333** [comment]
> I bumped version to `1.0.1` but why does this PR keep building with `0.12.11`

where did you see that?

**Comment by chenrui333** [comment]
btw, you can use `brew style --fix xx` to fix the syntax issue

**Comment by nbari** [comment]
> > I bumped version to `1.0.1` but why does this PR keep building with `0.12.11`
> 
> where did you see that?

I think that title changed when I accepted the review on the changes, got confused, sorry.

Thanks for the tip on the `style --fix`, I submitted changes,

**Comment by chenrui333** [comment]
you need to have the commit msg as `ssh-vault 1.0.1`, I think it is good to go after that. 👍

**Review by chenrui333 — APPROVED** [review]
Thanks! 💯

**Comment by nbari** [comment]
> you need to have the commit msg as `ssh-vault 1.0.1`, I think it is good to go after that. 👍

Done, many thanks, subsequent requests can be done with the `: mislav/bump-homebrew-formula-action@v3.1` action right ?

**Comment by nbari** [comment]
mm, some of the tests were truncated, how to retry?

**Comment by chenrui333** [comment]
> > you need to have the commit msg as `ssh-vault 1.0.1`, I think it is good to go after that. 👍
> 
> Done, many thanks, subsequent requests can be done with the `: mislav/bump-homebrew-formula-action@v3.1` action right ?

yeah, that would work

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**